src/test/java/EventSchedulerTest.java

```diff
@@ -0,0 +1,191 @@
+import data_gateway.CalendarManager;
+import data_gateway.EventReader;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import services.event_creation.CalendarEventModel;
+import services.event_from_task_creation.EventScheduler;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class EventSchedulerTest {
+    EventScheduler eventScheduler;
+    CalendarManager manager;
+    List<EventReader> events = new ArrayList<>();
+
+    @BeforeEach
+    void setup() {
+        manager = new MockCalendarManager(events);
+        eventScheduler = new EventScheduler(manager);
+
+        Set<String> tags1 = new HashSet<>();
+        tags1.add("CSC207");
+        tags1.add("Quiz");
+
+        Set<LocalDate> dates1 = new HashSet<>();
+        dates1.add(LocalDate.of(2021, 11, 25));
+        dates1.add(LocalDate.of(2021, 11, 27));
```

Comment:
would be nice to have these Date/DateTime/Time literals to be named instead. It's a little bit more work but it helps get a better and faster grasp at what the test is actually doing.

Reply:
So should I rename the variables to reflect the actual dates it contains (e.g. nov25Nov27)?

Reply:
Yep personally I think that would help make it more readable.
Ultimately this is for readability, so if adding variables like nov25 makes it feel less readable, feel free to not make the change.

Reply:
Got it

Reply:
I see you've changed some of the literals. Did you choose to not do the rest for a specific reason? As an example, I see a lot of November 25 dates, so it might be nice to have a more global `nov25 = LocalDate.of(2021, 11, 25)` and then use `nov25.atTime(insert time)` for LocalDateTimes